src/Metric.jl

```diff
@@ -1,50 +1,202 @@
+@doc doc"""
+    Metric
+
+Abstract type for the pseudo-Riemannian metric tensor $g$, a family of smoothly
+    varying inner products on the tangent space. See [`inner`](@ref).
+"""
 abstract type Metric end
 
+@doc doc"""
+    RiemannianMetric <: Metric
+
+Abstract type for Riemannian metrics, a family of positive definite inner
+    products. The positive definite property means that for $v \in T_x M$,
+    the inner product $g(v,v) > 0$.
+"""
 abstract type RiemannianMetric <: Metric end
 
-abstract type LorentzianMetric <: Metric end
+"""
+    LorentzMetric <: Metric
+
+Abstract type for Lorentz metrics, which have a single temporal dimension.
+"""
+abstract type LorentzMetric <: Metric end
```

Comment:
Do you want `LorentzMetric`s to have the signature +++...+- or ---..-+?

Reply:
I would opt for `++...+-`, but that is (just) a design decision that should be well documented then.

Reply:
I'm not certain which is best, (++...+-), (--...-+), (-+...++), (+-...--). All 4 conventions seem common. I wonder if it's too crazy to support multiple conventions via dispatch.

Reply:
I doubt I'll ever use it so all I care about is that the final decision should be documented.

Reply:
I might use it, but it's really a matter of taste (mine is `++...+-`), but as long as its documented, I am also fine with exactely one, because its easier to implement.

Reply:
I'm comfortable defaulting to this convention. Like anywhere else, a user is free to change the convention by implementing a special `MPoint` that changes it.